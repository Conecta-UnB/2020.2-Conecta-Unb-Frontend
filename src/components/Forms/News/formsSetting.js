import * as Yup from 'yup';

export const initialValues = {
  nome: '',
  matricula: '',
  email: '',
  senha: '',
  telefone: '',
};

export const SignupSchema = Yup.object().shape({
  titulo: Yup.string().required('É necessário informar um título'),
  autor: Yup.string().required('É necessário indicar um autor'),
  imagem: Yup.string(),
  conteudo: Yup.string().required('É necessário indicar o conteúdo'),
});

export const handleSubmit = async (values, setSubmitting, createNews) => {
  setSubmitting(false);
  await createNews({
    titulo: values.titulo,
    autor: values.autor,
    imagem: values.imagem,
    conteudo: values.conteudo,
  });
};
