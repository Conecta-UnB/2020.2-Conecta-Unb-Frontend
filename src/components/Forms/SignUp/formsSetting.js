import * as Yup from 'yup';

export const initialValues = {
  nome: '',
  matricula: '',
  email: '',
  senha: '',
  telefone: '',
};

export const SignupSchema = Yup.object().shape({
  nome: Yup.string().required('Required'),
  matricula: Yup
    .string()
    .length(9, 'Insira a matricula sem o /, total de 9 números')
    .matches(/^\d+$/, 'A matricula deve ser um número')
    .required('É necessário indicar uma matricula válida'),
  email: Yup
    .string()
    .email('Email inváilido')
    .required('É necessário indicar um email'),
  telefone: Yup
    .string()
    .length(11, 'O telefone deve ter 11 digitos')
    .matches(/^\d+$/, 'O telefone deve ser um número')
    .required('É necessário indicar um número'),
  senha: Yup
    .string()
    .min(6, 'A senha deve ter pelo menos 6 dígitos')
    .required('É necessário criar uma senha'),

});

export const handleSubmit = async (values, setSubmitting, createUser) => {
  setSubmitting(false);
  await createUser({
    nome: values.nome,
    matricula: values.matricula,
    email: values.email,
    senha: values.senha,
    telefone: values.telefone,
  });
};
