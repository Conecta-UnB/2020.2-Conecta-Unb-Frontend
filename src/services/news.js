export async function createNews(Host, news, matricula) {
  console.log(news, matricula);
  try {
    const response = await fetch(`${Host}/api/news/create`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        autor: news.autor,
        titulo: news.titulo,
        conteudo: news.conteudo,
        id_user: matricula,
      }),
    });
    const data = await response.json();
    console.log('Success: ', data);
    return data;
  } catch (error) {
    console.log('error ', error);
    return error;
  }
}

export async function getAllNews(Host) {
  try {
    const response = await fetch(`${Host}/api/news`, {
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
    });
    const data = await response.json();
    console.log('Success: ', data);
    return data;
  } catch (error) {
    console.log('error ', error);
    return error;
  }
}
