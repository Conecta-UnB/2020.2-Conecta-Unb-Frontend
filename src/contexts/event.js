import React, { createContext, useState } from 'react';
import * as EventService from '../services/event';

export const EventContext = createContext({});

// eslint-disable-next-line react/prop-types
export default function EventProvider({ children }) {
  const [allEvents, setAllEvents] = useState('');
  const [events] = useState('');
  const Host = 'http://localhost:3333';

  async function createEvent(event) {
    console.log('ola', event);
    const response = await EventService.createEvent(Host, event);
    return response;
  }
  async function updateEvent(event) {
    console.log('ola', event);
    const response = await EventService.updateEvent(Host, event);
    return response;
  }

  async function deleteEvent(event) {
    console.log('ola', event);
    const response = await EventService.deleteEvent(Host, event);
    return response;
  }

  async function readEvent(event) {
    console.log('ola', event);
    const response = await EventService.readEvent(Host, event);
    return response;
  }

  async function readAllEvents() {
    const response = await EventService.readAllEvents(Host);
    setAllEvents(response.data);
    console.log(response.data);
    console.log('estou aqui');
    return response;
  }

  return (
    <EventContext.Provider
      value={{
        events,
        allEvents,
        createEvent,
        updateEvent,
        deleteEvent,
        readAllEvents,
        readEvent,
      }}
    >
      {children}
    </EventContext.Provider>
  );
}
