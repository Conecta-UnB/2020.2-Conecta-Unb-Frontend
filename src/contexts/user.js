import React, { createContext } from 'react';
import * as UserService from '../services/user';

export const UserContext = createContext({});

// eslint-disable-next-line react/prop-types
export default function UserProvider({ children }) {
  // const [user, setUser] = useState({});
  const Host = 'http://localhost:3333';
  async function createUser(user) {
    console.log('ola', user);
    const response = await UserService.createUser(Host, user);
    return response;
  }

  async function login(user) {
    const response = await UserService.login(Host, user);
    return response;
  }

  return (
    <UserContext.Provider
      value={{
        createUser,
        login,
      }}
    >
      {children}
    </UserContext.Provider>
  );
}
