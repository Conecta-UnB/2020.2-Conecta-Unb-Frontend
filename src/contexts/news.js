import React, { createContext, useContext, useState } from 'react';
import * as NewsService from '../services/news';
import { UserContext } from './user';

export const NewsContext = createContext({});

// eslint-disable-next-line react/prop-types
export default function NewsProvider({ children }) {
  const { user } = useContext(UserContext);
  const Host = 'http://localhost:3333';
  const [allNews, setAllNews] = useState('');

  async function createNews(newsInfo) {
    console.log('news info =', newsInfo, user);
    console.log('user info =', user);
    const response = await NewsService.createNews(Host, newsInfo, user.matricula);
    return response;
  }

  async function getAllNews() {
    const response = await NewsService.getAllNews(Host);
    setAllNews(response.data.news);
    console.log(response.data.news);
    return response;
  }

  return (
    <NewsContext.Provider
      value={{
        createNews,
        getAllNews,
        allNews,
      }}
    >
      {children}
    </NewsContext.Provider>
  );
}
