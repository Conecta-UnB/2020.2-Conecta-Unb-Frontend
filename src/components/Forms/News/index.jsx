import React, { useContext } from 'react';
import {
  Formik, Form, Field, ErrorMessage,
} from 'formik';
import { initialValues, SignupSchema, handleSubmit } from './formsSetting';
import './styles.css';
import { NewsContext } from '../../../contexts/news';

export default function NewsForm() {
  const { createNews } = useContext(NewsContext);

  return (
    <div className="containerNewsForm">
      <div>
        <Formik
          initialValues={initialValues}
          onSubmit={(values, { setSubmitting }) => {
            handleSubmit(values, setSubmitting, createNews);
          }}
          validationSchema={SignupSchema}
        >
          {({ isSubmitting }) => (
            <Form>
              <Field name="titulo" placeholder="Título" />
              <ErrorMessage name="titulo" component="div" className="errorField" />
              <Field name="autor" placeholder="Autor" />
              <ErrorMessage name="autor" component="div" className="errorField" />
              <Field name="imagem" placeholder="Imagem" />
              <ErrorMessage name="imagem" component="div" className="errorField" />
              <Field type="textarea" name="conteudo" placeholder="Conteúdo" />
              <ErrorMessage name="conteudo" component="div" className="errorField" />
              <button type="submit" disabled={isSubmitting}>
                Postar
              </button>
            </Form>
          )}
        </Formik>
      </div>
    </div>
  );
}
