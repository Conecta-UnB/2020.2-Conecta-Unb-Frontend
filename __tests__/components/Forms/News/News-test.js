import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import {
  render, waitFor, screen,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NewsForms from '../../../../src/components/Forms/News';
import * as formSettings from '../../../../src/components/Forms/News/formsSetting';
import { NewsContext } from '../../../../src/contexts/news';

describe('testando news creation forms', () => {
  test('must match snapshot', () => {
    const tree = render(<NewsForms />);
    expect(tree).toMatchSnapshot();
  });

  test('test if the form works', async () => {
    const createNews = jest.fn();
    formSettings.handleSubmit = jest.fn();
    render(
      <NewsContext.Provider value={{ createNews }}>

        <MemoryRouter initialEntries={[{ pathname: '/login' }]}>
          <NewsForms />
        </MemoryRouter>
      </NewsContext.Provider>,
    );
    userEvent.type(screen.getByPlaceholderText('Título'), 'Asfalto na FGA');
    userEvent.type(screen.getByPlaceholderText('Autor'), 'Alguem');
    userEvent.type(screen.getByPlaceholderText('Imagem'), 'aaaaaaaaaaaaaaa');
    const a = screen.getByPlaceholderText('Conteúdo');
    userEvent.type(a, 'Finalmente a FGA tem asfalto');
    userEvent.click(screen.getByRole('button', { name: /Postar/i }));
    await waitFor(() => {
      expect(formSettings.handleSubmit).toHaveBeenCalledTimes(1);
      expect(formSettings.handleSubmit).toHaveBeenCalledWith({
        titulo: 'Asfalto na FGA',
        autor: 'Alguem',
        imagem: 'aaaaaaaaaaaaaaa',
        conteudo: 'Finalmente a FGA tem asfalto',
      },
      expect.any(Function),
      expect.any(Function));
    });
  });
});
