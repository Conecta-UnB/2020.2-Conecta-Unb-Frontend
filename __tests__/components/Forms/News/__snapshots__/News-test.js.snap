// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando loginforms deve ser igual a snapshot 1`] = `
<DocumentFragment>
  <div
    class="containerNewsForm"
  >
    <div>
      <form>
        <input
          autocomplete="off"
          placeholder="Título"
          type="text"
        />
        <textarea
          placeholder="Conteúdo"
        />
        <button
          type="submit"
        >
          Postar
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
